.github/workflows/ruff.yml

```diff
@@ -0,0 +1,16 @@
+# https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+name: ruff
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  #  branches:
+  #    - main
```

Comment:
I think these comments can be removed.  Also, the main branch here is called `master`.

Maybe just:
```
on: [push, pull_request]
```

Reply:
I find that `on: [push, pull_request]` will make duplicate runs for people who have write permissions to the repo (i.i. maintainers) so I tend to use
```yaml
on:
  push: [master]
  pull_request: [master]
```
but let's keep it simple given the speed of these runs.